### ISSUE TITLE
Wasm example not working out

### ISSUE THREAD

**Issue by inzanez** [ajrcarey/pdfium-render#95]
I just tried running the WASM example and I always get `Initialization of pdfium-render failed!`. I used to build everything manually so far with emscripten which is quite a pain. Now it feels more like magic, but...can't tell why that wouldn't work.

**Comment by inzanez** [comment]
@ajrcarey hm,...I have been using the latest build of Pdfium ([V5668](https://github.com/paulocoutinhox/pdfium-lib/releases/tag/5668))

Switching to the version that you mentioned fixed the issue. Is there any versioning hint or similar that would point out what version works? Not sure, maybe I missed something here. But great solution anyway! :-)

**Comment by ajrcarey** [comment]
No, there isn't any versioning hint, and annoyingly Pdfium itself doesn't provide any function to return the library version :/

I wouldn't describe this as a "solution" exactly, more a work-around. The problem is that the emscripten-provided malloc() function has changed in some way, although glancing at the function table for the latest Pdfium WASM build I don't immediately see the problem.

Probably something subtle changed in the emscripten packaging, and it's just a question of tracking down what it was and figuring out how to work around it in `pdfium-render`. I'll leave this issue open to work on that. But for now you have a work-around: use an older library version!

**Comment by inzanez** [comment]
Perfect, many thanks for that insight! I hope I can join you soon on digging into that issue. Sounds really interesting, but not that easy, remembering the whole emscripten around :-)

**Comment by ajrcarey** [comment]
The emscripten wrappers are in the pdfium.js file. Best guess is that the definition of Module._malloc() has changed in there in a recent WASM build of Pdfium compared to older releases. I'll try to take a look at it over the weekend.

**Comment by ajrcarey** [comment]
The `pdfium.js` file from V5407 contains the following export:

`var _malloc=Module["_malloc"]=createExportWrapper("malloc")`

But the `pdfium.js` file from V5668 has the following:

`var _malloc=createExportWrapper("malloc")`

The missing `Module["_malloc"]=` is the source of the problem. Changing this to match V5407's `pdfium.js` resolves the problem. The same adjustments need to be made for the `free` function.

**Comment by inzanez** [comment]
@ajrcarey I would assume that must be updated in `src/wasm.rs`, right? The link to `_malloc` and `_link`...

**Comment by ajrcarey** [comment]
The `createExportWrapper()` function in emscripten's `pdfium.js` looks like it returns a reference to `Module["asm"][<function name>]`. It might be possible to use that internally in `src/wasm.rs` instead of `Module["_malloc"]`, etc.

**Comment by ajrcarey** [comment]
Adjusted `PdfiumRenderWasmState::bind_to_pdfium()` to use `Module["asm"]["malloc"]` and `Module["asm"]["free"]` in the event `Module["_malloc"]` and `Module["_free"]` are not available directly. Updated README. Ready to release as part of crate version 0.8.8.

**Comment by ajrcarey** [comment]
As there have been no further comments and I believe the issue is now resolved, I am closing the issue and publishing release 0.8.8. Feel free to re-open if you do not believe the issue is resolved in release 0.8.8.
